Assets/ComethSDK/Scripts/Services/EoaFallbackService.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Text;
+using System.Threading.Tasks;
+using ComethSDK.Scripts.HTTP;
+using ComethSDK.Scripts.Tools;
+using ComethSDK.Scripts.Tools.Signers;
+using ComethSDK.Scripts.Types;
+using Nethereum.Signer;
+using UnityEngine;
+
+namespace ComethSDK.Scripts.Services
+{
+	public static class EoaFallbackService
+	{
+		public static async Task<EncryptionData> EncryptEoaFallback(string walletAddress,
+			string privateKey, string salt)
+		{
+			var encodedWalletAddress = Encoding.UTF8.GetBytes(walletAddress);
+			var encodedSalt = Encoding.UTF8.GetBytes(salt);
+
+			var encryptionKey =
+				await CryptoService.Pbkdf2(encodedWalletAddress, encodedSalt, Constants.PBKDF2_ITERATIONS);
+
+			var encodedPrivateKey = Encoding.UTF8.GetBytes(privateKey);
+
+			var iv = CryptoService.GetRandomIV();
+
+			var encryptedPrivateKey = await CryptoService.EncryptAESCBC(encryptionKey, iv, encodedPrivateKey);
+
+			return new EncryptionData
+			{
+				encryptedPrivateKey = Convert.ToBase64String(encryptedPrivateKey),
+				iv = Convert.ToBase64String(iv)
+			};
+		}
+
+		public static async Task<string> DecryptEoaFallback(
+			string walletAddress, byte[] encryptedPrivateKey, byte[] iv, string salt)
+		{
+			var encodedWalletAddress = Encoding.UTF8.GetBytes(walletAddress);
+			var encodedSalt = Encoding.UTF8.GetBytes(salt);
+
+			var encryptionKey =
+				await CryptoService.Pbkdf2(encodedWalletAddress, encodedSalt, Constants.PBKDF2_ITERATIONS);
+
+			var privateKey = await CryptoService.DecryptAESCBC(encryptionKey, iv, encryptedPrivateKey);
+
+			return Encoding.UTF8.GetString(privateKey);
+		}
+
+		public static async Task<Signer> GetSigner(API api, string provider, string walletAddress,
+			string encryptionSalt)
+		{
+			var storagePrivateKey = await GetSignerLocalStorage(walletAddress, encryptionSalt);
+
+			if (string.IsNullOrEmpty(storagePrivateKey))
+				throw new Exception("New Domain detected. You need to add that domain as signer.");
+
+			var storageSigner = new Signer(new EthECKey(storagePrivateKey));
+
+			var isOwner = await SafeService.IsSigner(storageSigner.GetAddress(), walletAddress, provider, api);
+
+			if (!isOwner) throw new Exception("New Domain detected. You need to add that domain as signer.");
```

Comment:
Is the error related to this error message?
I don't know what the owner check does in this context. However, if that's not an implementation user, make sure it's explicitly reported to the user.

Reply:
For this one i just went with the same logic as the typescript SDK.
In this context the owner check is used to know if the current signer is an Owner of the Safe, if not we throw an exception to tell the user that he needs to add the address as an Signer for the Safe